### PR TITLE
TypeScript definitions added

### LINKED ISSUE #8: TypeScript definitions missing
Hey,

could you please add TypeScript definitions?  Maybe to `@types/gulp-run-command`?

bildschirmfoto 2018-06-19 um 14 36 48

Thanks!

### PR DESCRIPTION
**Pull request by zhaparoff**
Fixes #8

### CODE REVIEW

**Comment by zhaparoff** [comment]
@Klathmon, @AndrewLeedham

Hi guys,

I've found this plugin very usefull for complex gulp builds.
But my gulp config is in TS.
Could you please review and approve a PR?
Or kindly let me know if anything should be fixed prior to that.

Regards.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+* import run from 'gulp-run-command';
+* 
+* gulp.task('clean', run('rm -rf build'));
+* gulp.task('build', ['clean'], run('babel index.js --out-file index.es5.js', {
+*   env: { NODE_ENV: 'production' }
+* })); 
+*/
+declare function run(command: string | string[], options?: Options): () => Promise<void>;
+
+export = run;
```
Comment (AndrewLeedham) [inline index.d.ts]: I could be wrong here but should this not be exporting as default. As the es5 bundle has `exports.default = run` on line 23 here https://unpkg.com/browse/gulp-run-command@0.0.9/index.es5.js as apposed to `exports = run`?
```suggestion
export default run;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+* 
+* @example 
+* import gulp from 'gulp';
+* import run from 'gulp-run-command';
+* 
+* gulp.task('clean', run('rm -rf build'));
+* gulp.task('build', ['clean'], run('babel index.js --out-file index.es5.js', {
+*   env: { NODE_ENV: 'production' }
+* })); 
+*/
```
Comment (AndrewLeedham) [inline index.d.ts]: Looks like all of the `*`'s should be indented by 1 space like in the options interface.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+* Async function to let gulp know it's complete when the promise resolves
+* 
+* @example 
+* import gulp from 'gulp';
+* import run from 'gulp-run-command';
+* 
+* gulp.task('clean', run('rm -rf build'));
+* gulp.task('build', ['clean'], run('babel index.js --out-file index.es5.js', {
+*   env: { NODE_ENV: 'production' }
+* }));
```
Comment (AndrewLeedham) [inline index.d.ts]: Perhaps this can be in a ` ```js ` block like the 'some advanced features' sample here: https://microsoft.github.io/tsdoc/

**Review by AndrewLeedham — CHANGES_REQUESTED** [review]

**Comment by AndrewLeedham** [comment]
@zhaparoff this looks great. I am not a maintainer or contributor for this repo, but I thought I'd leave some comments 😄

File: index.d.ts (hunk 3 above)
Comment (zhaparoff) [reply index.d.ts]: Probably you are right, but until TSDoc isn't an accepted standard, I don't believe using its features is reasonable. TSDoc advanced features could be unsupported by many IDEs other than VSCode at the moment.

Comment (AndrewLeedham) [reply index.d.ts]: Fair point. I suppose my thought process is somewhat bias as I use VSCode day to day.

File: index.d.ts (hunk 2 above)
Comment (zhaparoff) [reply index.d.ts]: Yeah, for some reason VSCode don't want to format that automatically. Fixed.

**Review by AndrewLeedham — APPROVED** [review]

File: index.d.ts (hunk 1 above)
Comment (zhaparoff) [reply index.d.ts]: Good point. Fixed.

**Comment by zhaparoff** [comment]
@AndrewLeedham, thanks for your review and feedback.

**Comment by AndrewLeedham** [comment]
> @AndrewLeedham, thanks for your review and feedback.

No worries. Hopefully @Klathmon can review and it can get merged in 😄

**Comment by Klathmon** [comment]
Hey thanks for the contribution! I'm happy to merge this in.

I'll also make a new release now (i'm assuming it's needed), probably as version `0.0.10`

Also, if either of you want to takeover as maintainer of this package (as i'm clearly not doing much of anything with it any more), feel free to shoot me an email at gregbenner1@gmail.com and we can chat.

**Pull request merged**